### PR TITLE
Fix deletion of non-uploaded languages

### PR DESCRIPTION
**Pull request by orangejenny**
PR into https://github.com/dimagi/commcare-hq/pull/23658

Fixes a bug where when a single-language file is uploaded, all translations in *other* languages are deleted.

(product team, ignore this)

### CODE REVIEW

**Comment by orangejenny** [comment]
Added another few commits to fix a bug where nodes were being deleted from forms when a single-language upload didn't provide a translation. Nodes should only be deleted if the upload is providing all languages and there are no values for any of them.

File: corehq/apps/translations/app_translations/upload_form.py [hunk 1]
```diff
@@ -158,13 +158,13 @@ def _add_or_remove_translations(self, lang, row):
         label_id = row['label']
         translations = self._get_translations_for_row(row, lang)
         translation_node = self.itext.find("./{f}translation[@lang='%s']" % lang)
-        keep_value_node = any(v for k, v in translations.items())
+        keep_value_node = any(v for k, v in translations.items()) or set(self.langs) != set(self.app.langs)
         text_node = self._get_text_node(translation_node, label_id)
         for trans_type, new_translation in translations.items():
-            if not new_translation:
+            if not new_translation and len(self.langs) > 1:
```
Comment (mkangia) [inline corehq/apps/translations/app_translations/upload_form.py]: If there is actually just one lang in the app this block would never run?

File: corehq/apps/translations/app_translations/upload_form.py [hunk 2]
```diff
@@ -158,13 +158,13 @@ def _add_or_remove_translations(self, lang, row):
         label_id = row['label']
         translations = self._get_translations_for_row(row, lang)
         translation_node = self.itext.find("./{f}translation[@lang='%s']" % lang)
-        keep_value_node = any(v for k, v in translations.items())
+        keep_value_node = any(v for k, v in translations.items()) or set(self.langs) != set(self.app.langs)
```
Comment (mkangia) [inline corehq/apps/translations/app_translations/upload_form.py]: i think we should put this in a variable with a nice name like
`all_app_langs_in_upload = set(self.langs) != set(self.app.langs)` 

Also better to keep this before the translations iterations?

File: corehq/apps/translations/app_translations/upload_form.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/translations/app_translations/upload_form.py]: Yes - if there's one language and the translation is blank, there aren't any other languages to fall back to, so this block is irrelevant.

File: corehq/apps/translations/app_translations/upload_form.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/translations/app_translations/upload_form.py]: I was thinking about setting a flag at the class level to indicate whether this is single-sheet or multi-sheet - adding in 31aecb9c9314389055242c32512379ef5d196b9e. Doing that makes this clearer, I think.

Also swapped the order of the conditions.

**Review by mkangia — APPROVED** [review]

**Review by emord — APPROVED** [review]

**Pull request merged**